### ISSUE TITLE
Application Registration

### ISSUE THREAD

**Issue by kami-delaroz** [microsoft/Microsoft365DSC#3403]
I'm trying to run the **Export-M365DSCConfiguration** command using the _application_ authentication.
I created an App Registration, gave it API Permissions of MicrosoftGraph for User, Group and Organization ReadWrite.All.
I also generated a secret and copied its value.
I tried to run
Export-M365DSCConfiguration -Components @("AADGroup") -ApplicationId "xxx" -ApplicationSecret "yyy" -TenantId "zzz.onmicrosoft.com"
and received 
```
Get-MgOrganization : The identity of the calling application could not be established.
At C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.23.614.1\modules\M365DSCUtil.psm1:1513 char:13
```
The relevant area in the code generates an error: _Error retrieving Organizational information: Missing Organization.Read.All permission._

----

I also tried :

$ApplicationId = "xxx"
$SecuredPassword = "yyy"
$tenantID = "zzz"

$SecuredPasswordPassword = ConvertTo-SecureString -String $SecuredPassword -AsPlainText -Force

$ClientSecretCredential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $ApplicationId, $SecuredPasswordPassword
Connect-MgGraph -TenantId $tenantID -ClientSecretCredential $ClientSecretCredential

and got:

`Connect-MgGraph : A parameter cannot be found that matches parameter name 'ClientSecretCredential'.
At line:1 char:37`

**Comment by andikrueger** [comment]
The error message is kind of expected. This issue already raised in #3292. I would close this issue in favor of the other.

**Comment by andikrueger** [comment]
![image](https://github.com/microsoft/Microsoft365DSC/assets/15608729/0d8d5df9-d12d-4a95-a329-96096e8d17f6)

**Comment by kami-delaroz** [comment]
Thanks.
As I wrote, I already gave Organization-Read permissions. so I'm not sure this is what I was missing

**Comment by andikrueger** [comment]
This must have been the missing part.

**Comment by andikrueger** [comment]
Is the application registration in the same tenant, that you try to export?

**Comment by kami-delaroz** [comment]
Indeed.
I also tried to gain access using certificate. upon running
`Connect-MgGraph -ClientID "e78b2368-edf4-44d7-84e0-af464d8180fe" -TenantId "MSFT.onmicrosoft.com" -CertificateThumbprint "6BDA00E3D87CEAAC7312E88F41DEED29FBA1AF4B"`
I got
`Welcome To Microsoft Graph!`
but I got the same error when running
`Export-M365DSCConfiguration -Components @("AADGroup") -ApplicationId "e78b2368-edf4-44d7-84e0-af464d8180fe" -CertificateThumbprint "6BDA00E3D87CEAAC7312E88F41DEED29FBA1AF4B" -TenantId "MSFT.onmicrosoft.com"`

**Comment by NikCharlebois** [comment]
I am not able to replicate in our environments. I created brand new App registrations and ensured only the same permissions as what you have in your above screenshot were granted. 

Are you getting an error when running:

``` powershell
Connect-MgGraph -ClientID "e78b2368-edf4-44d7-84e0-af464d8180fe" -TenantId "MSFT.onmicrosoft.com" -CertificateThumbprint "6BDA00E3D87CEAAC7312E88F41DEED29FBA1AF4B"
Get-MgOrganization -ErrorAction 'Stop'
```

**Comment by kami-delaroz** [comment]
I got the same error.

I tried to create a new APP registration, gave it only user and org permissions, and tried authenticate with either secret or certificate, still same error. screenshots attached.

I'm using [this guide ](https://learn.microsoft.com/en-us/azure/active-directory/develop/howto-create-self-signed-certificate) for generating the certificate

![Screen Shot 2023-06-25 at 9 18 00](https://github.com/microsoft/Microsoft365DSC/assets/66998687/cd85c0db-e3d4-42a2-bb4b-4d69b169b901)

![Screen Shot 2023-06-25 at 9 19 02](https://github.com/microsoft/Microsoft365DSC/assets/66998687/a767af54-bd01-4567-b8db-7cc2239ad3b5)

<img width="1505" alt="Screen Shot 2023-06-25 at 9 52 55" src="https://github.com/microsoft/Microsoft365DSC/assets/66998687/e7ccde73-e97f-4a24-89a1-dd25cbd4ed5d">

**Comment by andikrueger** [comment]
Would you mind giving this cmdlet a try: https://microsoft365dsc.com/user-guide/cmdlets/Update-M365DSCAzureAdApplication/

This should create a functioning app registration with the suitable permissions.

**Comment by NikCharlebois** [comment]
We cannot replicate even using the same instructions you shared to create a self signed certificate. Can you please try on a different machine/environment to see if you are able to replicate? This looks like it is specific to your environment unfortunately.

**Comment by kami-delaroz** [comment]
Hi, getting back to my old issue, would really appreciate some help here.
I created a new windows vm (version 2022), opened pwsh in administration level, and run:
```
Install-Module Microsoft365DSC -Force
Update-M365DSCDependencies
Export-M365DSCConfiguration -Components @("AADGroup") -Credential $Credential
```
so far so good.
then I tried to create a new app registration. for some reason the app isn't shown under **Owned applications**, but under **All applications**. so I tried the following steps for both the new app registration and an old existing one:
1. create a certificate according to [this](https://learn.microsoft.com/en-us/azure/active-directory/develop/howto-create-self-signed-certificate) documentation and upload it
2. give it **a lot of** app permissions I came across relevant issue discussion (screenshot included). I understand most are irrelevant but..
3. create an app secret
4. run both Application based and Certificate based export command. got errors on both:

for `Export-M365DSCConfiguration -Components @("AADGroup") -ApplicationId $ApplicationId -ApplicationSecret $ApplicationSecret -TenantId $TenantId` I got:
<img width="947" alt="Screen Shot 2023-08-02 at 13 56 02" src="https://github.com/microsoft/Microsoft365DSC/assets/66998687/8185bf02-ace2-4b94-9209-3a523d009437">

for `Export-M365DSCConfiguration -Components @("AADGroup") -ApplicationId $ApplicationId -CertificateThumbprint $CertificateThumbprint -TenantId $TenantId` I got:
<img width="952" alt="Screen Shot 2023-08-02 at 13 55 43" src="https://github.com/microsoft/Microsoft365DSC/assets/66998687/55bc5c7c-88f4-4b3d-82e5-f824cc442032">


Those are the App Permissions I gave the App Registration:
<img width="921" alt="Screen Shot 2023-08-02 at 14 11 00" src="https://github.com/microsoft/Microsoft365DSC/assets/66998687/cff6adfb-63c7-4bc9-89fc-af35a17e6ff3">


---------------------

Any explanation / step by step instructions of getting the export command to work with either Application based or Certificate based authentication is welcome
